src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -1141,32 +1165,15 @@ private static TextEdit BuildRevertedEdit(TextEdit provisionalTextEdit)
                 return null;
             }
 
-            var synchronized = await _documentSynchronizer.TrySynchronizeVirtualDocumentAsync(
-                request.HostDocument.Version, virtualDocumentSnapshot, rejectOnNewerParallelRequest: false, cancellationToken);
-
+            var synchronized = await _documentSynchronizer.TrySynchronizeVirtualDocumentAsync(request.HostDocument.Version, virtualDocumentSnapshot, rejectOnNewerParallelRequest: false, cancellationToken);
             if (!synchronized)
             {
                 return null;
             }
 
-            var hoverParams = new TextDocumentPositionParams()
-            {
-                TextDocument = new TextDocumentIdentifier()
-                {
-                    Uri = projectedUri,
-                },
-                Position = request.ProjectedPosition,
-            };
-
-            var textBuffer = virtualDocumentSnapshot.Snapshot.TextBuffer;
-            var response = await _requestInvoker.ReinvokeRequestOnServerAsync<TextDocumentPositionParams, VSInternalHover?>(
-                textBuffer,
-                Methods.TextDocumentHoverName,
-                languageServerName,
-                hoverParams,
-                cancellationToken).ConfigureAwait(false);
-
-            return response?.Response;
+            return new DelegationRequestDetails(languageServerName, projectedUri, virtualDocumentSnapshot.Snapshot.TextBuffer);
         }
+
+        private record DelegationRequestDetails(string LanguageServerName, Uri ProjectedUri, ITextBuffer TextBuffer);
```

Comment:
Could get away with a record struct here

Reply:
I am not good at structs, and so tend to avoid them. We will be lost without someone actually smart on this team :)

Reply:
This made me have to put `.Value` everywhere and now I'm sad :(